grafanalib/cloudwatch.py

```diff
@@ -0,0 +1,194 @@
+            "expression": "",
+            "highResolution": False,
+            "id": "",
+            "metricName": self.metricName,
+            "namespace": self.namespace,
+            "period": self.period,
+            "refId": self.refId,
+            "region": self.region,
+            "returnData": False,
+            "statistics": self.statistics
+        }.update(self.dimensions)
+
+    def __checkParameters(self):
+        self.__checDimensions()
+        self.__checkNamespace()
+        self.__checkRegion()
+
+    def __checkDimensions(self):
+        if self.dimensions == {}:
+            raise Exception(
+                'You need to define a valid non empty dimensions dict variable'
+            )
+
+    def __checkNamespace(self):
+        if not (self.namespace in validNamespaces):
+            raise Exception(
+                '"{}" is not valid Cloudwatch namespace'.format(
+                    self.namespace)
+            )
+
+    def __checkRegion(self):
+        if not (self.region in validRegions):
+            raise Exception(
+                '"{}" is not valid AWS region'.format(
+                    self.region)
+            )
+
+
+validNamespaces = [
+    "AWS/ApiGateway",
+    "AWS/AppStream",
+    "AWS/AppSync",
+    "AWS/Athena",
+    "AWS/Billing",
+    "AWS/ACMPrivateCA",
+    "AWS/CloudFront",
+    "AWS/CloudHSM",
+    "AWS/CloudSearch",
+    "AWS/Logs",
+    "AWS/CodeBuild",
+    "AWS/Cognito",
+    "AWS/Connect",
+    "AWS/DataSync",
+    "AWS/DMS",
+    "AWS/DX",
+    "AWS/DocDB",
+    "AWS/DynamoDB",
+    "AWS/EC2",
+    "AWS/EC2Spot",
+    "AWS/AutoScaling",
+    "AWS/ElasticBeanstalk",
+    "AWS/EBS",
+    "AWS/ECS",
+    "AWS/EFS",
+    "AWS/ElasticInference",
+    "AWS/ApplicationELB",
+    "AWS/ELB",
+    "AWS/NetworkELB",
+    "AWS/ElasticTranscoder",
+    "AWS/ElastiCache",
+    "AWS/ElastiCache",
+    "AWS/ES",
+    "AWS/ElasticMapReduce",
+    "AWS/MediaConnect",
+    "AWS/MediaConvert",
+    "AWS/MediaPackage",
+    "AWS/MediaTailor",
+    "AWS/Events",
+    "AWS/FSx",
+    "AWS/FSx",
+    "AWS/GameLift",
+    "AWS/Glue",
+    "AWS/Inspector",
+    "AWS/IoT",
+    "AWS/IoTAnalytics",
+    "AWS/ThingsGraph",
+    "AWS/KMS",
+    "AWS/KinesisAnalytics",
+    "AWS/Firehose",
+    "AWS/Kinesis",
+    "AWS/KinesisVideo",
+    "AWS/Lambda",
+    "AWS/Lex",
+    "AWS/ML",
+    "AWS/Kafka",
+    "AWS/AmazonMQ",
+    "AWS/Neptune",
+    "AWS/OpsWorks",
+    "AWS/Polly",
+    "AWS/QLDB",
+    "AWS/Redshift",
+    "AWS/RDS",
+    "AWS/Robomaker",
+    "AWS/Route53",
+    "AWS/SageMaker",
+    "AWS/SDKMetrics",
+    "AWS/DDoSProtection",
+    "AWS/SES",
+    "AWS/SNS",
+    "AWS/SQS",
+    "AWS/S3",
+    "AWS/SWF",
+    "AWS/States",
+    "AWS/StorageGateway",
+    "AWS/Textract",
+    "AWS/Transfer",
+    "AWS/Translate",
+    "AWS/TrustedAdvisor",
+    "AWS/NATGateway",
+    "AWS/TransitGateway",
+    "AWS/VPN",
+    "AWS/WorkMail",
+    "AWS/WorkSpaces",
+]
+
+validRegions = [
```

Comment:
Would it make sense to use something like https://stackoverflow.com/a/38451512 here?